src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -157,8 +158,42 @@ public boolean hasOverlappingSession(Session target) {
         for (Student student : internalList) {
             List<Session> sessionList = student.getListOfSessions();
             for (Session session : sessionList) {
-                if (session.isOverlapping(target)) {
-                    return true;
+                if (session instanceof RecurringSession) {
+                    RecurringSession recurringSession = (RecurringSession) session;
+                    if (recurringSession.isOverlapping(target)) {
```

Comment:
Possible long slap here, the other group might state that this is a code quality violation, so do take note.

Reply:
Will do in 1.4! Thanks!!